src/zarr/array.py

```diff
@@ -232,25 +222,21 @@ async def _read_chunk(
         out_selection: SliceSelection,
         out: np.ndarray,
     ) -> None:
-        chunk_spec = self.metadata.get_chunk_spec(chunk_coords)
+        chunk_spec = self.metadata.get_chunk_spec(chunk_coords, self.order)
```

Comment:
here is where we tell the signal to codec pipeline what the order of the output chunk is.